hal-sim/hal_impl/functions.py

```diff
@@ -856,7 +856,24 @@ def freeDIOPort(dioPortHandle):
 
 def allocateDigitalPWM(status):
```

Comment:
@virtuald / @auscompgeek can you confirm that this is correct logically and the correct way to go about this? I think I interpreted the C code correctly, but not 100% sure.